manifests/repos.pp

```diff
@@ -22,19 +37,62 @@
       $package_key   = $docker::package_key
       $package_repos = $docker::package_repos
 
+      if ( $facts['os']['distro']['id'] == 'Debian' and versioncmp($facts['os']['distro']['release']['major'],'10') >= 0 ) or ( $facts['os']['distro']['id'] == 'Ubuntu' and versioncmp($facts['os']['distro']['release']['major'],'22') >= 0 ) { # lint:ignore:140chars
+        # fix deprecated apt-key warnings
+        if $curl_ensure {
+          ensure_packages(['curl'])
+        }
+        if $gpg_ensure {
+          ensure_packages(['gpg'])
+        }
+        if $keyring_force_update {
+          exec { 'Remove Docker-GPG-Key':
+            path    => '/bin/',
+            cwd     => '/tmp',
+            command => "rm ${keyring}",
```

Comment:
Are you sure that you want to remove the keyring from /tmp directory?

Reply:
Since $keyring is an absolute path the cwd should have no negative effect on the rm.

Reply:
OK. Makes sense.